Notifications: localise start of the week

simison opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Part of DOTCOM-13034

## Proposed Changes

* Update "Plain Text" to "Plain text" for copy consistency across UIs (sentence casing)
<img width="238" alt="Screenshot 2025-05-13 at 12 46 22" src="https://github.com/user-attachments/assets/3c2db628-3528-4399-b0b4-667ec2bffdac" />
<img width="239" alt="Screenshot 2025-05-13 at 12 46 13" src="https://github.com/user-attachments/assets/3f407881-b0c2-447f-8b03-90fbd1dfc719" />


* Update "HTML" to "Visual (HTML)" to be clearer to those non-technical
<img width="249" alt="Screenshot 2025-05-13 at 12 45 11" src="https://github.com/user-attachments/assets/3cf800b9-dff6-4046-bd13-807c2b70ce7e" />
<img width="251" alt="Screenshot 2025-05-13 at 12 45 01" src="https://github.com/user-attachments/assets/cb6ba1f5-bc21-4f2c-9c8b-741616744750" />


* Star week by Monday, Saturday or Sunday in weekday selector based on locale.

Uses native `new Intl.Locale( locale ).getWeekInfo()` API from browsers if available via a new utility function in `@automattic/i18n-utils`. When not available (basically just Firefox), defaults to "Monday".

<img width="590" alt="Screenshot 2025-05-12 at 15 16 45" src="https://github.com/user-attachments/assets/735a6414-11aa-4839-b412-fc5141c7667a" />

**Before** (all locales, all browsers)
<img width="250" alt="Screenshot 2025-05-12 at 15 16 40" src="https://github.com/user-attachments/assets/901009ae-4d40-4f08-99b2-1242b76f8dc0" />

**After**
Spanish
<img width="374" alt="Screenshot 2025-05-13 at 11 59 55" src="https://github.com/user-attachments/assets/1564fc57-7308-4de8-8431-7bb16e0e2916" />

English
<img width="281" alt="Screenshot 2025-05-13 at 12 00 41" src="https://github.com/user-attachments/assets/20dcadab-f369-46e3-975c-6186aa94cbd1" />

Brazilian Portuguese
<img width="276" alt="Screenshot 2025-05-13 at 12 01 19" src="https://github.com/user-attachments/assets/928e35ef-9552-4499-bc3f-54ce7ab07964" />

Arabic
<img width="248" alt="Screenshot 2025-05-13 at 12 40 11" src="https://github.com/user-attachments/assets/8482481a-0080-40bf-9745-cac92285de76" />


Firefox on all locales
<img width="267" alt="Screenshot 2025-05-13 at 12 05 01" src="https://github.com/user-attachments/assets/c8c7d622-6e68-403d-a583-f66a2a1334a8" />





## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

*

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to `/me/notifications/subscriptions`
* See "Email delivery window" dropdown in different locales, different browsers.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

simison committed:
Notifications: put Sunday last in daylist

enejb commented:
Should Sunday move be updated based on the `start_of_week` setting?
 
See general options.  
![Screenshot 2025-05-12 at 8 31 28 AM](https://github.com/user-attachments/assets/b9fb1e3f-0339-469e-83da-05da2b664c60)

simison commented:
@enejb that's a site specific setting, but we could respect `new Intl.Locale( locale ).weekInfo`

edanzer reviewed (dismissed):
Tests as expected. I see there's a discussion about making the day order conditional on site / users / Intl api. I'd be fine with either way. Approving in any case as the PR does what's intended and it's a simple change.

simison committed:
Retreive first day of the week from browser API

Uses vanilla API for retrieving value from `Locale.prototype.getWeekInfo()`, or the currently more supported `Locale.prototype.weekInfo`, or if neither is available defaults to "Monday". As of 05/2025, Firefox does not support either.

simison commented on packages/i18n-utils/src/date-utils.ts:
```diff
@@ -125,3 +125,37 @@ export const getNumericDateTimeString = ( timestamp: number, locale = 'en' ) =>
+ * @param {number} defaultFirstDayOfWeek Default first day of week if browser API is not available or locale is incorrect.
+ * @returns {number} First day of week 1–7, 1 being Monday and 7 being Sunday.
+ */
+export const getNumericFirstDayOfWeek = ( locale: string, defaultFirstDayOfWeek: number = 1 ) => {
```
@Automattic/i18n I wanted to check with you that adding a simple utility like this is fine. The purpose isn't so significant that it doesn't justify a heavy polyfill, but it's a nice UX touch that helps to localize UI.

dlind1 replied:
The approach looks good to me. It's unfortunate that there is no support for Firefox yet, but at least all of the other major browsers are supported.

simison committed:
Add Arabic Saturday to the test

simison committed:
List via function to accommodate Saturday

simison committed:
Fix test

simison committed:
HTML → Visual (HTML)

simison commented:
Updated to take locale into account, including Arabic locales which start week by Saturday.

Updated description with details.

simison committed:
Simplify utility by not allowing default to be set

Let's not complicate until someone actually needs default to be set manually.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17466534

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @simison for including a screenshot in the description! This is really helpful for our translators.

dhasilva commented on client/me/notification-settings/reader-subscriptions/index.jsx:
```diff
@@ -111,6 +112,53 @@ class NotificationSubscriptions extends Component {
+			case 1:
+				return (
+					<>
+						<option value="1">{ translate( 'Monday' ) }</option>
+						<option value="2">{ translate( 'Tuesday' ) }</option>
+						<option value="3">{ translate( 'Wednesday' ) }</option>
+						<option value="4">{ translate( 'Thursday' ) }</option>
+						<option value="5">{ translate( 'Friday' ) }</option>
+						<option value="6">{ translate( 'Saturday' ) }</option>
+						<option value="7">{ translate( 'Sunday' ) }</option>
+					</>
+				);
+			// Saturday
+			case 6:
+				return (
+					<>
+						<option value="6">{ translate( 'Saturday' ) }</option>
+						<option value="7">{ translate( 'Sunday' ) }</option>
+						<option value="1">{ translate( 'Monday' ) }</option>
+						<option value="2">{ translate( 'Tuesday' ) }</option>
+						<option value="3">{ translate( 'Wednesday' ) }</option>
+						<option value="4">{ translate( 'Thursday' ) }</option>
+						<option value="5">{ translate( 'Friday' ) }</option>
+					</>
+				);
+			// Sunday
+			case 7:
+				return (
+					<>
+						<option value="7">{ translate( 'Sunday' ) }</option>
+						<option value="1">{ translate( 'Monday' ) }</option>
+						<option value="2">{ translate( 'Tuesday' ) }</option>
+						<option value="3">{ translate( 'Wednesday' ) }</option>
+						<option value="4">{ translate( 'Thursday' ) }</option>
+						<option value="5">{ translate( 'Friday' ) }</option>
+						<option value="6">{ translate( 'Saturday' ) }</option>
+					</>
+				);
+		}
+	}
```
This can be simplified to avoid repetition with something like:

```suggestion
	renderLocalizedWeekdayOptions() {
		const { translate, locale } = this.props;
		const startOfWeek = getNumericFirstDayOfWeek( locale );

		const weekDays = [
			{ value: '1', label: translate( 'Monday' ) },
			{ value: '2', label: translate( 'Tuesday' ) },
			{ value: '3', label: translate( 'Wednesday' ) },
			{ value: '4', label: translate( 'Thursday' ) },
			{ value: '5', label: translate( 'Friday' ) },
			{ value: '6', label: translate( 'Saturday' ) },
			{ value: '7', label: translate( 'Sunday' ) },
		];

		// Rotate the array based on startOfWeek
		const rotatedWeekdays = [
			...weekDays.slice( startOfWeek - 1 ),
			...weekDays.slice( 0, startOfWeek - 1 ),
		];

		return (
			<>
				{ rotatedWeekdays.map( ( { value, label } ) => (
					<option key={ value } value={ value }>
						{ label }
					</option>
				) ) }
			</>
		);
	}
```

simison replied:
I considered that but went with simple, quick to read and grok approach since it's basically similar amount of code.

In yours it's quicker to make centralized changes to options and I don't mind way or the other, so I'll apply your suggestion. :-)

simison committed:
Use slice to re-arrange weekdays

Co-authored-by: Douglas Henri <dhasilva@users.noreply.github.com>

simison committed:
Changelog typo

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17466534).
